spec/bolt_spec/run_spec.rb

```diff
@@ -107,18 +107,25 @@ def root_config
       } }
     end
 
+    def agent_inventory
+      { 'groups' => conn_inventory['groups'],
```

Comment:
So the reason these tests are failing is that `apply_manifest` calls `apply_prep`, which used to see that an agent was installed and do nothing. Now it tries to install the agent, but doesn't know to run try as `root` instead of as `bolt`.

This change is modifying the inventory for the entire test to use an inventory that says that there's currently an agent installed on the targets, even when there's not (like after we uninstall the agent on line 120). I think a better approach to this (and kind of the intended approach) is to set the agent feature in the inventory after running the `puppet_agent::install` task on line 121, then remove the feature after uninstalling the agent on line 128.

Reply:
You additionally shouldn't have to use a modified inventory for any of the tasks since those can't check for target features.